### ISSUE TITLE
Add group description in the editor

### ISSUE THREAD

**Issue by snouk-z** [godotengine/godot-proposals#5846]
### Describe the project you are working on

A game which uses node groups.

### Describe the problem or limitation you are having in your project

I can't provide contextual informations about a group to help identify its purpose or specificities.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the possibility to write a description for a group.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- Implementing, in the Group Editor window, a text area to edit and view the description for the currently selected group.
- Having a tooltip that displays the description when the mouse is hovering the group name in the Node tab

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It is editor-related.

### Is there a reason why this should be core and not an add-on in the asset library?

Sounds more like a primary feature than an add-on.

**Comment by YuriSizov** [comment]
Just to be clear, this is about node groups, correct? This is a bit confusing because there is also a current discussion in the community about documenting property groups.

**Comment by snouk-z** [comment]
@YuriSizov That's correct. It is about allowing the user to add a description to the node group he created.

**Comment by DarkMessiah** [comment]
Here I have implemented a description for "global" groups :) https://github.com/godotengine/godot/pull/60965
